File: day1/part2.exs
[left, right] =
  File.read!("./input")
  |> String.split("\n", trim: true)
  |> Enum.map(fn x ->
    String.split(x, ~r{\s+}, trim: true) |> Enum.map(&String.to_integer/1) |> List.to_tuple()
  end)
  |> Enum.unzip()
  |> Tuple.to_list()

freqs = Enum.frequencies(right)

left
|> Enum.reduce(0, fn x, acc ->
  acc + x * Map.get(freqs, x, 0)
end)
|> IO.puts()

File: day1/part2.ts
const f = Bun.file("./input");
const lines = (await f.text())
  .split("\n")
  .filter((l) => l !== "")
  .map((l) => l.split(/\s+/).map((n) => parseInt(n)));
let left: number[] = [];
let right: Map<number, number> = new Map();
for (const [l, r] of lines) {
  left.push(l);
  right.set(r, (right.get(r) ?? 0) + 1);
}
let sum = 0;
for (const l of left) {
  sum += l * (right.get(l) ?? 0);
}
console.log(sum);
